biggus/__init__.py

```diff
@@ -824,6 +832,47 @@ def masked_array(self):
         return array
 
 
+def _use_indexed_array(keys):
+    return ((isinstance(keys, tuple) and any(key is None for key in keys)) or
+            keys is None)
+
+
+class IndexedArray(Array):
+    def __init__(self, array, keys):
+        self.array = array
+        if not isinstance(keys, tuple):
+            keys = (keys,)
+        self.keys = tuple(keys)
+
+    @property
+    def dtype(self):
+        return self.array.dtype
+
+    @property
+    def shape(self):
+        return _sliced_shape(self.array.shape, self.keys)
+
+    def __getitem__(self, keys):
+        raise RuntimeError('unsupported operation')
```

Comment:
What's the timeline/priority, here? ... I understand you wanna get this nailed down to ship with the Iris v1.8.0 release?